internal/ingress/controller/store/endpointslice_test.go

```diff
@@ -91,4 +92,27 @@ func TestEndpointSliceLister(t *testing.T) {
 			t.Errorf("expected %v, error, got %v", endpointSlice.GetName(), eps[0].GetName())
 		}
 	})
+	t.Run("svc long name", func(t *testing.T) {
+		el := newEndpointSliceLister(t)
+		ns := "namespace"
+		svcName := "test-backend-http-test-http-test-http-test-http-test-http-truncated"
+		key := fmt.Sprintf("%s/%s", ns, svcName)
+		endpointSlice := &discoveryv1.EndpointSlice{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns,
+				Name:      "test-backend-http-test-http-test-http-test-http-test-http-bar88",
+				Labels: map[string]string{
+					discoveryv1.LabelServiceName: svcName,
+				},
+			},
+		}
+		el.Add(endpointSlice)
+		eps, err := el.MatchByKey(key)
+		if err != nil {
+			t.Errorf("unexpeted error %v", err)
+		}
+		if len(eps) != 1 {
+			t.Errorf("expected one slice %v, error, got %d slices", endpointSlice, len(eps))
+		}
```

Comment:
It'd also be good to assert that the 1 eps we expect is the eps we want (with name `test-backend-http-test-http-test-http-test-http-test-http-bar88` and namespace `ns`).